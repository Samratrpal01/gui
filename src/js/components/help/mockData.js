export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {},
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '01-01-2003'
  }
};
