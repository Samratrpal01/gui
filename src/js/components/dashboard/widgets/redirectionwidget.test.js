import React from 'react';

import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { undefineds } from '../../../../../tests/mockData';
import { render } from '../../../../../tests/setupTests';
import RedirectionWidget from './redirectionwidget';

describe('RedirectionWidget Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(<RedirectionWidget onClick={jest.fn} content="testlocation" />);
    const view = baseElement;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });

  it('works as intended', async () => {
    const content = 'test content';
    const submitCheck = jest.fn();
    render(<RedirectionWidget content={content} onClick={submitCheck} />);

    userEvent.click(screen.getByText(content));
    expect(screen.queryByText('redirected')).not.toBeInTheDocument();
    expect(submitCheck).toHaveBeenCalledTimes(1);
    userEvent.click(screen.getByText(content));
    expect(submitCheck).toHaveBeenCalledTimes(2);
  });
});
