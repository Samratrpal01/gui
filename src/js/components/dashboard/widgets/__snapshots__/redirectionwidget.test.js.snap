// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedirectionWidget Component renders correctly 1`] = `
<body>
  <div>
    <div
      class="widget flexbox centered"
    >
      <p
        class="muted"
        style="max-width: 200px;"
      >
        testlocation
      </p>
    </div>
  </div>
</body>
`;
